src/command_modules/azure-cli-monitor/azure/cli/command_modules/monitor/tests/test_monitor_diagnostic_settings.py

```diff
@@ -0,0 +1,71 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+from azure.cli.testsdk import ScenarioTest, JMESPathCheck, ResourceGroupPreparer, StorageAccountPreparer
+
+
+class TestMonitorDiagnosticSettings(ScenarioTest):
+    @ResourceGroupPreparer(location='southcentralus')
+    @StorageAccountPreparer(location='southcentralus')
+    def test_monitor_diagnostic_settings(self, resource_group, storage_account):
+        nsg_name = self.create_random_name(prefix='nsg', length=16)
+        self.cmd('network nsg create -n {} -g {}'.format(nsg_name, resource_group))
+
+        log_config = json.dumps([
+            {
+                "category": "NetworkSecurityGroupEvent",
+                "enabled": True,
+                "retentionPolicy": {"days": 0, "enabled": False}
+            },
+            {
+                "category": "NetworkSecurityGroupRuleCounter",
+                "enabled": True,
+                "retentionPolicy": {"days": 0, "enabled": False}
+            }
+        ])
```

Comment:
Users have to type this on the command line?...
Kind of verbose for them.

Reply:
You could load it from a JSON file with @{file}

Reply:
It is what we have today. And yes it is not ideal. I intended to further improve this in the next stage. I